skimage/filters/thresholding.py

```diff
@@ -389,3 +391,109 @@ def threshold_li(image):
             new_thresh = temp + tolerance
 
     return threshold + immin
+
+
+def threshold_mean(image):
+    """Return threshold value based on the of grayscale values.
+
+    Parameters
+    ----------
+    image : array
+        Grayscale input image.
+
+    Returns
+    -------
+    threshold : float
+        Upper threshold value. All pixels intensities that less or equal of
+        this value assumed as foreground.
+
+    References
+    ----------
+    .. [1] C. A. Glasbey, "An analysis of histogram-based thresholding
+        algorithms," CVGIP: Graphical Models and Image Processing,
+        vol. 55, pp. 532-537, 1993.
+
+    Examples
+    --------
+    >>> from skimage.data import camera
+    >>> image = camera()
+    >>> thresh = threshold_mean(image)
+    >>> binary = image > thresh
+    """
+    return np.mean(image.ravel())
+
+
+def threshold_triangle(image, nbins=256):
+    """Return threshold value based on triangle algorithm.
+
+    Parameters
+    ----------
+    image : array
+        Grayscale input image.
+    nbins : int, optional
+        Number of bins used to calculate histogram. This value is ignored for
+        integer arrays.
+
+    Returns
+    -------
+    threshold : float
+        Upper threshold value. All pixels intensities that less or equal of
+        this value assumed as foreground.
+
+    References
+    ----------
+    .. [1] Zack, G. W., Rogers, W. E. and Latt, S. A., 1977,
+       Automatic Measurement of Sister Chromatid Exchange Frequency,
+       Journal of Histochemistry and Cytochemistry 25 (7), pp. 741-753
+
+    Examples
+    --------
+    >>> from skimage.data import camera
+    >>> image = camera()
+    >>> thresh = threshold_triangle(image)
+    >>> binary = image > thresh
+    """
+    # nbins is ignored for interger arrays
+    # so, we recalculate the effective nbins.
+    hist, bin_centers = histogram(image.ravel(), nbins)
+    nbins = bin_centers[-1] - bin_centers[0] + 1
+
+    # Find peak and histogram extremities.
+    argpeak_height = np.argmax(hist)
+    peak_height = np.max(hist)
+    arglefth = np.where(hist>0)[0][0]
+    argrighth = np.where(hist>0)[0][-1]
```

Comment:
These variable names are too hard to read. Could you rename them, please or add some underscores?